src/main/encryptors/Encryptor.ts

```diff
@@ -21,18 +21,39 @@ export interface DecryptResult<T> {
   decryptedItem: T
 }
 
-export type EncryptedJayZItem<T, U extends keyof T> = ItemWithEncryptedFields<T, U> & {
-  __jayz__metadata: EncryptedItemMetadata<T, U>
+export type EncryptedJayZItem<T, U extends keyof T> = LegacyEncryptedJayZItem<T, U> | EncryptedJayZItemV1<T, U>
+
+export type LegacyEncryptedJayZItem<T, U extends keyof T> = ItemWithEncryptedFields<T, U> & {
+  __jayz__metadata: {
+    version: string
+    keyId?: string
+    nonce: Uint8Array
+    encryptedDataKey: Uint8Array
+    encryptedFieldNames: U[]
+  }
 }
 
-export type ItemWithEncryptedFields<T, U extends keyof T> = Omit<T, U> & {
-  [K in U]: Uint8Array
+export type EncryptedJayZItemV1<T, U extends keyof T> = Omit<T, U> & {
+  __jayz__metadata: {
+    metadataVersion: MetadataVersion.V1
+    encryptorId: string
+    keyId?: string
+    nonce: Uint8Array
+    encryptedDataKey: Uint8Array
+    encryptedFields: Uint8Array
+  }
+}
+
+export function isLegacyJayZItem<T, U extends keyof T>(
```

Comment:
This is a TypeScript [type-guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html) -- which is a function where if it returns true, TypeScript will "auto-cast" the input object to the specified type.